broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java

```diff
@@ -650,10 +652,13 @@ public void protectBroker() {
 
     public long headSlowTimeMills(BlockingQueue<Runnable> q) {
         long slowTimeMills = 0;
-        final Runnable peek = q.peek();
-        if (peek != null) {
-            RequestTask rt = BrokerFastFailure.castRunnable(peek);
-            slowTimeMills = rt == null ? 0 : this.messageStore.now() - rt.getCreateTimestamp();
+        Optional<RequestTask> op = q.stream()
```

Comment:
I am a little confusing about this. We are using `java.util.concurrent.LinkedBlockingQueue` for this `q`, and it should be thread safe. Can you give a detailed example for this change to work?

Reply:
Thanks for your review.

It has nothing to do with thread safety, There is to reduce thread contention, `q` is polled by thread in `BrokerController#sendMessageExecutor` and will be peeked by thread in `BrokerController#scheduledExecutorService`  per second as shown below:

```
this.scheduledExecutorService.scheduleAtFixedRate(new Runnable() {
                @Override
                public void run() {
                    try {
                        BrokerController.this.printWaterMark();
                    } catch (Throwable e) {
                        log.error("printWaterMark error.", e);
                    }
                }
            }, 10, 1, TimeUnit.SECONDS);
```
You can look at the issue #2516 , get some context info first.

Reply:
OK, I see. Nice work